src/normalization.jl

```diff
@@ -106,11 +106,26 @@ function imsharpen(
 
     image_equalized_gray = Gray.(image_equalized)
 
-    image_smoothed = imfilter(image_equalized_gray, Kernel.gaussian(smoothing_param))
+    return unsharp_mask(image_equalized_gray, smoothing_param, intensity)
+end
+
+"""
+    unsharp_mask(image_equalized_gray, smoothing_param, intensity)
+
+Apply unsharp masking to an equalized grayscale image to enhance its sharpness.
 
-    image_sharpened =
-        image_equalized_gray .* (1 + intensity) .+ image_smoothed .* (-intensity)
-    return min.(max.(image_sharpened, 0.0), 1.0)
+# Arguments
+- `image_equalized_gray: The input grayscale image that has been equalized.
+- `smoothing_param::Int`: The pixel radius for Gaussian blurring (typically between 1 and 10).
+- `intensity: The amount of sharpening to apply. Higher values result in more pronounced sharpening.
+
+# Returns
+The sharpened grayscale image with values clipped between 0.0 and 1.0.
+"""
+function unsharp_mask(image_equalized_gray, smoothing_param, intensity)
```

Comment:
does the gray image have to be equalized for the algorithm to work? if not, I'd rename this argument to be `image_gray`. If so, can you say why?

Reply:
No, it doesn't. That name was chosen as a reminder that in the intended use case workflow this operation is applied to an image that has been equalized and has values either in [0, 1] or [0, 255]. @danielmwatkins what do you think?

Reply:
I think that the more general name is good! Since this one is applied to a grayscale image naming the argument image_gray makes sense.